### PR TITLE
Return a `Rack::BodyProxy` from the `Rails::Rack::Logger` monkey patch

### PR DESCRIPTION
**Pull request by ghiculescu**
This should return the same object that Rails is returning. See https://github.com/rails/rails/blob/8d1b8e870c1c42859132c8d51ceff0ac72cbe34b/railties/lib/rails/rack/logger.rb#L38

### CODE REVIEW

File: lib/lograge/rails_ext/rack/logger.rb [hunk 1]
```diff
@@ -12,7 +12,8 @@ class Logger
       # Overwrites Rails 3.2 code that logs new requests
       def call_app(*args)
         env = args.last
-        @app.call(env)
+        status, headers, body = @app.call(env)
+        [status, headers, ::Rack::BodyProxy.new(body) {} ]
```
Comment (ghiculescu) [inline lib/lograge/rails_ext/rack/logger.rb:16]:
```suggestion
        # needs to have same return type as the Rails builtins being overridden, see https://github.com/roidrage/lograge/pull/333
        [status, headers, ::Rack::BodyProxy.new(body) {} ]
```

**Comment by iloveitaly** [comment]
@ghiculescu could you rebase on master and add an entry to the changelog?

**Comment by ghiculescu** [comment]
Done, did you also want commits squashed or are you fine to use the github squasher?

**Comment by sigra** [comment]
@iloveitaly @ghiculescu 👋 

I'm not sure, but it looks like [Rack::BodyProxy#close](https://github.com/rack/rack/blob/master/lib/rack/body_proxy.rb#L29 ) always calls block in ensure part. But in this PR the block is `nil`, so I have `Read: #<NoMethodError: undefined method :call for nil:NilClass>` for each request. It doesn't affect user work, but this spam to stdout doesn't seem right.

Check the original implementation in Rails — https://github.com/rails/rails/blob/v6.1.4.4/railties/lib/rails/rack/logger.rb#L38. They use block.

Also, I checked all gems in my gemset and found out that ALL of them use block.
image

P.S. I've copied block logic from original Rails implementation and it solves my issue — https://github.com/Salesap/lograge/commit/d7ec60e78133b64bb164d4d66eb9f76887ae7391

**Comment by iloveitaly** [comment]
@sigra interesting! Mind submitting a PR to change our use of `BodyProxy`?

**Pull request merged**